signature/cose/envelope.go

```diff
@@ -497,27 +496,32 @@ func parseProtectedHeaders(protected cose.ProtectedHeader, signerInfo *signature
 	signerInfo.SignatureAlgorithm = sigAlg
 
 	// populate signerInfo.SignedAttributes.SigningScheme
-	signingSchemeString, ok := protected[headerLabelSigningScheme].(string)
-	if !ok {
-		return &signature.InvalidSignatureError{Msg: "invalid signingScheme"}
-	}
+	// headerLabelSigningScheme header has already been checked by
+	// validateCritHeaders() at the beginning of this function.
+	signingSchemeString := protected[headerLabelSigningScheme].(string)
 	signingScheme := signature.SigningScheme(signingSchemeString)
 	signerInfo.SignedAttributes.SigningScheme = signingScheme
-
-	// populate signerInfo.SignedAttributes.SigningTime
 	signingTimeLabel, ok := signingSchemeTimeLabelMap[signingScheme]
 	if !ok {
 		return &signature.InvalidSignatureError{Msg: "unsupported signingScheme: " + signingSchemeString}
 	}
-	signingTime, err := parseTime(protected[signingTimeLabel])
+
+	// generate RawProtected CBOR map
+	cborMap, err := generateRawProtectedCBORMap(rawProtected)
```

Comment:
`cborMap` is too generic. How about `headerMap` or `headers`?

Also, you might want to update the comment to be something like `parse CBOR map from raw protected header for tag validation`.